src/components/Tabset/index.js

```diff
@@ -48,10 +52,24 @@ export default class Tabset extends Component {
     }
 
     componentDidMount() {
-        this.updateButtonsVisibility();
         window.addEventListener('resize', this.updateButtonsVisibility);
     }
 
+    componentDidUpdate(prevProp) {
+        const { tabsetChildren } = this.state;
+        const { isFirstTime } = this;
+        const { children } = this.props;
+        const isAllChildrenRegistered = tabsetChildren.length > 0 && children.length === tabsetChildren.length;
```

Comment:
areAllChildrenRegistered
also review that first condition is necessary, if not remove it